modules/heat_conduction/src/materials/FunctionPathEllipsoidHeatSource.C

```diff
@@ -0,0 +1,68 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FunctionPathEllipsoidHeatSource.h"
+
+registerMooseObject("HeatConductionApp", FunctionPathEllipsoidHeatSource);
+
+template <>
+InputParameters
+validParams<FunctionPathEllipsoidHeatSource>()
+{
+  InputParameters params = Material::validParams();
+  params.addRequiredParam<Real>("power", "laser power");
+  params.addParam<Real>("efficiency", 1, "process efficiency");
+  params.addRequiredParam<Real>("rx", "transverse ellipsoid axe");
+  params.addRequiredParam<Real>("ry", "longitudinal ellipsoid axe");
+  params.addRequiredParam<Real>("rz", "depth ellipsoid axe");
+  params.addParam<Real>("factor", 1, "scaling factor");
+  params.addParam<FunctionName>("function_x", "0", "The x component heating spot travel path");
+  params.addParam<FunctionName>("function_y", "0", "The y component heating spot travel path");
+  params.addParam<FunctionName>("function_z", "0", "The z component heating spot travel path");
```

Comment:
Would it be better to say something like this for the description:

The x component of the center of the heating spot as a function of time